android/src/main/java/cloud/pace/plugins/cloudsdk/CloudSDK.kt

```diff
@@ -1,40 +1,139 @@
 package cloud.pace.plugins.cloudsdk
 
+import cloud.pace.plugins.cloudsdk.EnumUtils.searchEnum
 import cloud.pace.sdk.PACECloudSDK
-import cloud.pace.sdk.utils.AuthenticationMode
-import cloud.pace.sdk.utils.Configuration
-import cloud.pace.sdk.utils.Environment
+import cloud.pace.sdk.api.API
+import cloud.pace.sdk.api.geojson.GeoJSONAPI.geoJSON
+import cloud.pace.sdk.api.geojson.generated.request.geoJSON.GetBetaGeojsonPoisAPI
+import cloud.pace.sdk.api.geojson.generated.request.geoJSON.GetBetaGeojsonPoisAPI.getBetaGeojsonPois
+import cloud.pace.sdk.appkit.AppKit
+import cloud.pace.sdk.utils.*
 import com.getcapacitor.*
 
 @NativePlugin
 class CloudSDK : Plugin() {
+    private var cofuStationFetchRunning: Boolean = false
+
     @PluginMethod
     fun setup(call: PluginCall) {
         val apiKey = call.getString(API_KEY)
 
         if (apiKey == null) {
-            call.reject("Missing Api Key")
+            call.reject("Failed setup: Missing Api Key")
             return
         }
 
         val callAuthenticationMode = call.getString(AUTHENTICATION_MODE) ?: WEB
         val callEnvironment = call.getString(ENVIRONMENT) ?: PRODUCTION
 
-        val authenticationMode = AuthenticationMode.valueOf(callAuthenticationMode)
-        val environment = Environment.valueOf(callEnvironment)
+        val authenticationMode = try {
+            AuthenticationMode.valueOf(callAuthenticationMode)
+        } catch (e: Exception) {
+            AuthenticationMode.WEB
+        }
+
+        val environment = searchEnum(Environment::class.java, callEnvironment)
+                ?: Environment.PRODUCTION
 
-        val configuration: Configuration = Configuration(
+        val configuration = Configuration(
                 clientAppName = CLIENT_APP_NAME,
                 clientAppVersion = CLIENT_APP_VERSION,
                 clientAppBuild = CLIENT_APP_BUILD,
                 apiKey = apiKey,
-                authenticationMode = authenticationMode
+                authenticationMode = authenticationMode,
+                environment = environment
         )
 
         PACECloudSDK.setup(context, configuration)
         call.resolve()
     }
 
+    @PluginMethod
+    fun listAvailableCoFuStations(call: PluginCall) {
+        if (cofuStationFetchRunning) {
+            call.reject("Failed listAvailableCoFuStations: Already running")
+            return
+        }
+
+        val countries = call.getArray(COUNTRIES)
+        if (countries == null)
+            call.reject("Failed listAvailableCoFuStations: Missing countries value")
```

Comment:
This shouldn't be a failure. Requesting CoFu stations for all countries is a valid request.

@patrickniepel I missed this in the iOS wrapper.

Reply:
Updated

Reply:
@EmDee Will adjust this with the next issue